### ISSUE TITLE
Pylint does not escape strings before outputting them in messages

### ISSUE THREAD

**Issue by GideonBear** [pylint-dev/pylint#7874]
### Bug description

```py
var == '\n'
```
I confirmed this happens in `magic-value`, but I don't know if there are other errors that this can apply to.
Maybe it should use `repr` instead of `str` for representing values?
This also happens with other characters:
- `'\r'`: `'. (magic-value-comparison) R2004: Consider using a named constant or an enum instead of '`
- Any color code (for example `'\033[92m'`): Everything after `instead of '` is colored green

### Configuration

```ini
[MAIN]
load-plugins=pylint.extensions.magic_value
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
R2004: Consider using a named constant or an enum instead of '
'. (magic-value-comparison)
```

### Expected behavior

```
R2004: Consider using a named constant or an enum instead of '\n'. (magic-value-comparison)
```

### Pylint version

```shell
pylint 2.16.0-dev
astroid 2.12.13
Python 3.9.15 (main, Nov 11 2022, 13:33:56) 
[GCC 9.4.0]
```

**Comment by clavedeluna** [comment]
Can reproduce!

**Comment by GideonBear** [comment]
https://github.com/PyCQA/pylint/blob/8c789cc9184f880bdd282ea877c8ee932c7bfe28/pylint/extensions/magic_value.py#L65-L75
What's the best way to handle this? Should it just use `repr`?

**Comment by Pierre-Sassoulas** [comment]
Not sure; but possibly as we're trying more and more to add code suggestions in messages.
